src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -1176,6 +1178,10 @@ protected String checkJavaVersion(final PrintStream logger, String javaCommand,
         return null;
     }
 
+    private double requiredJavaVersion() {
+        return Double.valueOf("1." + JDK_MAJOR_VERSION);
```

Comment:
How about base this on a minimal version that is supported by jenkins version running on master (provided we can get that requirement at runtime)? Or at least use the version running on master as the minimum.